### DISCUSSION TITLE
Iterative reporting data preparation using digital field forms

### DISCUSSION THREAD

**Discussion by NewGraphEnvironment** [NewGraphEnvironment/dff-2022#154] [Announcements]
Its all a bit confusing to discuss.  Coming together though.  Here is a shot at defining the workflow concept:

1. R to read gpkg fresh from field and back up to version controlled csv
2. R to clean initial form fresh from the field - update version controlled csv then burn to back to Q for QA and the odd manual input.
3. Decide on workflow improvement - (ex. read gpkg back to R to join chris_culvert_ids to an amalgamated comments column)
4. Burn to version controlled csv ready for copy paste to spreadsheet and back to gpkg
5. Rinse and repeat

**Comment by Mateo9569** [comment]
For step 4 when we convert back to geopackage, should we be overwriting the `data_field/2023/form_pscis_2023.gpkg` file each time? And then create a backup gpkg?

**Reply by NewGraphEnvironment** [reply]
We don't need to backup `form_pscis_2023.gpkg` to a gpkg as long as the original `form_pscis_2023.gpkg` is version controlled as `form_pscis_2023.csv` in the reporting repo.  We can add the version number from mergin to the csv repo commit too and be sure to sync to mergin before and after changes. 

We do however need to make sure that the coordinates are burned to the csv so that it can be remade as a geopackage.  We should have a `x` column and a `y` column (in addition to our populated standard `utm_zone`, `easting` and `northing` columns) and burn them in `albers 3005` to eliminate any possible issues related to utms from multiple utm zones.

**Reply by NewGraphEnvironment** [reply]
Hmmm.  I actually don't think that mergin is going to be able to deal with schema changes (ie. extra columns) to the `gpkg`.  Pretty sure geodiff https://github.com/MerginMaps/geodiff can't do it and we should confirm at the start. _Shouldn't_ matter though as long as we don't use `form_pscis_2023.gpkg` for actual input on the phone in the field and we do the versioning in csvs with git (git is better for 99% of use cases anyway).

**Reply by Mateo9569** [reply]
Should I re burn to a geopackage at the end of the pscis_tidy.R script? Even though there are some different columns and columns missing that are in the original `form_pscis_2023` that's in the Q project now?

**Reply by NewGraphEnvironment** [reply]
See lines 1:109+ in fiss_site_tidy.R but need to also burn to  `form_fiss_2023.csv https://github.com/NewGraphEnvironment/fish_passage_peace_2023_reporting/blob/076d9585c05f747292ce936e7670da346afa6d12/scripts/fiss_site_tidy.R#L109

`gpkg` has all columns always.  `pscis_tidy` really does 2 things and could perhaps be two separate scripts (`pscis_tidy` and `pscis_export`)
 

pscis_tidy
1. import `form_pscis_2023.gpkg` - add `site_id`, tidy up things like trib and Trib to be Tributary, populate null values, create new column for appended comments with moti id, etc. - then Burn to version controlled csv with everything there (`data/dff/form_pscis_2023.csv`) and back to `QGIS/data_field_2023/form_pscis_2023.gpkg`.  Rinse and repeat

pscis_export
2. trim up complete table so it has the right columns - burn to `data/dff/pscis_export.csv` ready for copy paste. That part is quite straight forward so could likely be a function (ie. `dff::dff_psics_export`) 

then... if something changed in Q or needs to change in R - read gpkg back to R - change (if needed in R) - Burn to version controlled csv with everything   (`form_pscis_2023.csv`), `form_pscis_2023.gpkg`, as well as `pscis_export.csv` specifically made ready for copy paste to spreadsheet
